test_runner/regress/test_tenant_size.py

```diff
@@ -0,0 +1,342 @@
+    # expected to use the "next gc" cutoff, so the small amounts should work
+
+    env = neon_env_builder.init_start()
+
+    tenant_id = env.initial_tenant
+    branch_name, timeline_id = env.neon_cli.list_timelines(tenant_id)[0]
+
+    http_client = env.pageserver.http_client()
+
+    collected_responses: List[Tuple[Lsn, Any]] = []
+
+    with env.postgres.create_start(branch_name, tenant_id=tenant_id) as pg:
+        with pg.cursor() as cur:
+            cur.execute("CREATE TABLE t0 (i BIGINT NOT NULL)")
+
+        batch_size = 100
+
+        i = 0
+        while True:
+            with pg.cursor() as cur:
+                cur.execute(
+                    f"INSERT INTO t0(i) SELECT i FROM generate_series({batch_size} * %s, ({batch_size} * (%s + 1)) - 1) s(i)",
+                    (i, i),
+                )
+
+            i += 1
+
+            current_lsn = wait_for_last_flush_lsn(env, pg, tenant_id, timeline_id)
+
+            size, _ = get_tenant_size(http_client, tenant_id)
+
+            if len(collected_responses) > 0:
+                last_size = collected_responses[-1][1]
+                if size == 0:
+                    assert last_size == 0
+                else:
+                    assert size > last_size
+
+            collected_responses.append((current_lsn, size))
+
+            if len(collected_responses) > 2:
+                break
+
+        while True:
+            with pg.cursor() as cur:
+                cur.execute(
+                    f"UPDATE t0 SET i = -i WHERE i IN (SELECT i FROM t0 WHERE i > 0 LIMIT {batch_size})"
+                )
+                updated = cur.rowcount
+
+            if updated == 0:
+                break
+
+            current_lsn = wait_for_last_flush_lsn(env, pg, tenant_id, timeline_id)
+
+            size, _ = get_tenant_size(http_client, tenant_id)
+            assert size > collected_responses[-1][1]
+            collected_responses.append((current_lsn, size))
+
+        while True:
+            with pg.cursor() as cur:
+                cur.execute(f"DELETE FROM t0 WHERE i IN (SELECT i FROM t0 LIMIT {batch_size})")
+                deleted = cur.rowcount
+
+            if deleted == 0:
+                break
+
+            current_lsn = wait_for_last_flush_lsn(env, pg, tenant_id, timeline_id)
+
+            size, _ = get_tenant_size(http_client, tenant_id)
+            # even though rows have been deleted, the size should not decrease
+            assert size > collected_responses[-1][1]
+            collected_responses.append((current_lsn, size))
+
+        with pg.cursor() as cur:
+            cur.execute("DROP TABLE t0")
+
+        current_lsn = wait_for_last_flush_lsn(env, pg, tenant_id, timeline_id)
+
+        size, _ = get_tenant_size(http_client, tenant_id)
+
+        assert size > collected_responses[-1][1]
+        collected_responses.append((current_lsn, size))
+
+    for lsn, size in collected_responses:
+        log.info(f"collected: {lsn}, {size}")
```

Comment:
Both locally and remotely the values are stable in the environment, but the values between local and remote differ:

```
--- local-collected
+++ remote-collected
@@ -1,10 +1,10 @@
-0/16C1198, 23728536
-0/16C2DF8, 23735800
-0/16CAB48, 23767880
-0/16D2AC0, 23800512
-0/16D5838, 23812152
-0/16D97F0, 23828464
-0/16DB268, 23835240
-0/16DCCE8, 23842024
-0/16DE628, 23848488
-0/16E2C00, 23866368
+0/16BC518, 23708952
+0/16BE178, 23716216
+0/16C5EB0, 23748272
+0/16CDE28, 23789176
+0/16D2C08, 23800840
+0/16D6BC0, 23817152
+0/16D8638, 23823928
+0/16DA0B8, 23830712
+0/16DB9E0, 23837152
+0/16DDF68, 23846760
```

Reply:
23866368 - 23846760 = 19608; adding a comment on the log.info's about this.